include/brickengine/input.hpp

```diff
@@ -102,28 +121,120 @@ class BrickInput {
                                                 time_to_wait_for_key.second = 0;
                                             }
                                         } else {
-                                            inputs[player_id][input_mapping[player_id][*input]] = true;
+                                            if(axis_map.count(*input))
+                                                inputs[player_id][input_mapping[player_id][*input]] += axis_map[*input];
+                                            else
+                                                   inputs[player_id][input_mapping[player_id][*input]] = true;
                                         }
                                         break;
                                     case SDL_KEYUP:
-                                        inputs[player_id][input_mapping[player_id][*input]] = false;
+                                        if(axis_map.count(*input))
+                                            inputs[player_id][input_mapping[player_id][*input]] -= axis_map[*input];
+                                        else
+                                            inputs[player_id][input_mapping[player_id][*input]] = false;
+                                        break;
                                     default:
                                         break;
                                 }
                             }
                         }
                     }
                     break;
+                case SDL_JOYAXISMOTION: {
+                    for(auto [player_id, controller] : player_controller_mapping) {
+                        if(e.jaxis.which == controller) {
+                            // For some reason you cannot get the axis of the current joyaxis motion. So we need to calculate both.
+
+                            // Y and X are reversed in a controller
+                            auto y_value = SDL_GameControllerGetAxis(controllers.at(controller), SDL_CONTROLLER_AXIS_LEFTX);
+                            auto x_value = SDL_GameControllerGetAxis(controllers.at(controller), SDL_CONTROLLER_AXIS_LEFTY);
+
+                            // Scale the value from the controller range to -1 and 1
+                            double old_range = 32767 + 32768;
+                            double new_range = 1 + 1;
+                            double new_value_x = (((x_value + 32768) * new_range) / old_range) - 1;
+                            double new_value_y = (((y_value + 32768) * new_range) / old_range) - 1;
+                             
+
+                            if(x_value < JOYSTICK_DEADZONE && x_value > -JOYSTICK_DEADZONE)
+                                new_value_x = 0;
+
+                            if(y_value < JOYSTICK_DEADZONE && y_value > -JOYSTICK_DEADZONE)
+                                new_value_y = 0; 
+                    
```

Comment:
Spacing is a bit odd and place a space between the "-" and the JOYSTICK_DEADZONE